packages/data/src/lib/dataview/SourceDataView.ts

```diff
@@ -0,0 +1,80 @@
+import { MapsDataviews as DataviewsApi } from '@carto/toolkit-maps';
+import { Layer, CARTOSource } from '@carto/toolkit-viz';
+import { defaultCredentials, WithEvents } from '@carto/toolkit-core';
+import { CartoDataViewError, dataViewErrorTypes } from './DataViewError';
+
+export abstract class SourceDataView extends WithEvents {
+  private _source: CARTOSource | Layer;
+  private _column: string;
+  protected dataviewsApi: DataviewsApi;
+
+  constructor(
+    source: CARTOSource | Layer,
+    column: string,
+    credentials = defaultCredentials
+  ) {
+    super();
+    validateParameters(source, column);
+
+    this._source = source;
+    this._column = column;
+    this.dataviewsApi = new DataviewsApi(this._getSourceName(), credentials);
+
+    this.registerAvailableEvents(['dataChanged', 'optionChanged']);
+  }
+
+  public set source(newSource: CARTOSource | Layer) {
+    validateParameters(newSource, this._column);
+
+    this._source = newSource;
+    this.emit('optionChanged');
+  }
+
+  public get column() {
+    return this._column;
+  }
+
+  public set column(newColumn: string) {
```

Comment:
Wouldn't a new column potentially retrigger the maps instantiation?

Reply:
Yup, that's the idea. In this version we are not "caching" the map instantiation yet so it'll be instantiated always but when we implement the caching mechanism changing the column will invalid the cache and trigger the map instantiation again.

Reply:
Ok, just to be clear, that map instantiation is one specific for the dataviews, unrelated to the one for the map itself right? If that's the case, how does t behave if we do a setSource in the Layer? I guess it would have to be retriggered too or should the user recreate the dataviews with the new Source?. 

Please think a bit about these variations, so we can have a more detailed list of tasks for the next PR

Reply:
Yes, the map instantiation is one specific for the dataviews.

About if the user does a `setSource`, that's a good point. ATM the user would have to recreate the dataviews with the new source but I think the option you suggested is better. We likely should add an event within Layer which is triggered when the user changed the source in order to instantiate the maps of dataview again.

Reply:
Or at least an error in the meantime...